templates/web/pages/product/product_page.tt.html

```diff
@@ -47,7 +47,7 @@
     <p>→ <a href="[% public_product_url %]">[% lang("product_page_on_the_public_database") %]</a></p>
 [% END %]
 
-[% IF (user_id.defined) && (robotoff_url.defined) && (robotoff_url.length > 0) %]
+[% IF (robotoff_url.defined) && (robotoff_url.length > 0) %]
```

Comment:
yes it's that easy, because robotoff takes care of computing device_id using url.

Reply:
so for a non logged in OFF website user, how is the device_id computed? is it a hash of the user ip address or something?

Reply:
yes it's the sha1 of user ip, see https://github.com/openfoodfacts/robotoff/blob/e46942f0055b54183ef5570b7c39215ed312309c/robotoff/app/api.py#L235